vpr/src/pack/cluster_legalizer.h

```diff
@@ -0,0 +1,478 @@
+    vtr::vector_map<AtomBlockId, LegalizationClusterId> atom_cluster_;
+
+    /// @brief Stores the NoC group ID of each atom block. Atom blocks that
+    ///        belong to different NoC groups can't be clustered with each other
+    ///        into the same clustered block.
+    vtr::vector<AtomBlockId, NocGroupId> atom_noc_grp_id_;
+
+    /// @brief Stats keeper for placement information during packing/clustering.
+    /// TODO: This should be a vector.
+    /// FIXME: This keeps the stats for each cluster type. This is fine within
+    ///        the clusterer, however it yields a limitation where two clusters
+    ///        of the same type cannot be constructed at the same time. This
+    ///        should stored per cluster.
+    t_cluster_placement_stats* cluster_placement_stats_ = nullptr;
+
+    /// @brief The utilization of external input/output pins during packing
+    ///        (between 0 and 1).
+    t_ext_pin_util_targets target_external_pin_util_;
+
+    /// @brief The max size of any molecule. This is used to allocate a dynamic
+    ///        array within the legalizer, and in its current form this is a bit
+    ///        expensive to calculate from the prepacker.
+    size_t max_molecule_size_;
+
+    /// @brief The max size a cluster could physically be. This is used to
+    ///        allocate dynamic arrays.
+    size_t max_cluster_size_;
+
+    /// @brief A vector of routing resource nodes within each logical block type
+    ///        [0 .. num_logical_block_types-1]
+    /// TODO: This really should not be a pointer to a vector... I think this is
+    ///       meant to be a vector of vectors...
+    std::vector<t_lb_type_rr_node>* lb_type_rr_graphs_ = nullptr;
+
+    /// @brief The total number of models (user + library) in the architecture.
+    ///        Used to allocate space in dynamic data structures.
+    size_t num_models_;
+
+    /// @brief The current legalization strategy of the cluster legalizer.
+    ClusterLegalizationStrategy cluster_legalization_strategy_;
+
+    /// @brief Controls whether the pin counting feasibility filter is used
+    ///        during clustering. When enabled the clustering engine counts the
+    ///        number of available pins in groups/classes of mutually connected
+    ///        pins within a cluster. These counts are used to quickly filter
+    ///        out candidate primitives/atoms/molecules for which the cluster
+    ///        has insufficient pins to route (without performing a full
+    ///        routing). This reduces packing run-time. This matches the packer
+    ///        option of the same name.
+    bool enable_pin_feasibility_filter_;
+
+    /// @brief The max size of the priority queue for candidates that pass the
+    ///        early filter legality test but not the more detailed routing
+    ///        filter. This matches the packer option of the same name.
+    int feasible_block_array_size_;
```

Comment:
When I scanned the code, this is also what I observed. Its the max size of some queue which is used to store the number of "feasible blocks" in a given pb. Its a very strange structure and I plan to make the array dynamic in another PR later (or raise an issue to do so), since it really does not make sense.